manifests/profile/hathitrust/apache/redirection.pp

```diff
@@ -0,0 +1,77 @@
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::profile::hathitrust::apache::redirection
+#
+# hathitrust.org redirection-only virtual hosts
+#
+# @example
+#   include nebula::profile::hathitrust::apache::redirection
+class nebula::profile::hathitrust::apache::redirection (
+  String $sdrroot,
+  Hash $default_access,
+  Array[String] $haproxy_ips,
+  Hash $ssl_params,
+  String $prefix,
+  String $domain,
+  Array[String] $alias_domains = []
+) {
+
+
+  $babel_servername   = "${prefix}babel.${domain}"
+  $catalog_servername = "${prefix}catalog.${domain}"
+  $www_servername     = "${prefix}www.${domain}"
+
+
+  ['babel', 'catalog', 'm', 'www'].each |String $vhost| {
+    $servername = "${prefix}${vhost}.${domain}"
+
+    apache::vhost { "${servername} non-ssl":
+      servername        => $servername,
+      docroot           => false,
+      port              => '80',
+      redirect_source   => '/',
+      redirect_status   => 'permanent',
+      redirect_dest     => "https://${servername}",
+      access_log_file   => "${vhost}/access.log",
+      access_log_format => 'combined',
+      error_log_file    => "${vhost}/error.log"
+    }
+
+    file { "/var/log/apache2/${vhost}":
+      ensure => 'directory',
+      mode   => '0755',
+      owner  => 'root',
+      group  => 'root',
+    }
+  }
+
+  apache::vhost { 'hathitrust canonical name redirection':
+    servername        => $domain,
+    docroot           => false,
+    port              => '80',
+    # TODO test me
```

Comment:
This is tested as of https://github.com/mlibrary/nebula/commit/add0610a6ae16d684bb1baed235b53aaf03a1cc8; can remove this TODO.